File: apps/webshop/frontend/src/types/item.ts
export interface Item {
  id: number
  name: string
  qualityRetentionDays: number
  purchaseUnit: number
  leadTimeDays: number
  supplierName: string
}

export interface ItemRequest {
  name: string
  qualityRetentionDays: number
  purchaseUnit: number
  leadTimeDays: number
  supplierName: string
}

File: apps/app/models/item.rb
class Item < ApplicationRecord
  belongs_to :supplier

  validates :name, presence: true
  validates :quality_retention_days, presence: true, numericality: { greater_than: 0 }
  validates :purchase_unit, presence: true, numericality: { greater_than: 0 }
  validates :lead_time_days, presence: true, numericality: { greater_than: 0 }
end
